### ISSUE TITLE
Multiline logical comparison indentation error

### ISSUE THREAD

**Issue by DavZim** [r-lib/lintr#2857]
When I use multiline logical conditions such as the following, lintr throws an error but this is the format that is automatically indented by RStudio (Ctrl + I) or by air.

```R
is.null(a) ||
  length(a) == 1 &&
  is.na(a)
```

A full reprex is here:

```R
code_lines <- "is.null(a) ||
  length(a) == 1 &&
  is.na(a)
"

writeLines(code_lines)
#> is.null(a) ||
#>   length(a) == 1 &&
#>   is.na(a)

lintr::lint(
  text = code_lines,
  linters = lintr::indentation_linter()
)
#> '<text>:3:2': style: [indentation_linter] Indentation should be 4 spaces but is 2 spaces.
#>   is.na(a)
#>  ^~~
```

I would expect this to pass as I "feel" it is indented correctly.

```R
packageVersion("lintr")
#> [1] ‘3.2.0’
```

**Comment by MichaelChirico** [comment]
I agree. See also #2007, #2034

**Comment by MichaelChirico** [comment]
This was not fixed by #3123 (which fixes #2007 and #2535); {styler} also agrees with air and `indentation_linter()`, so I would file this as a bug with RStudio. Following is an explanation from Gemini; I'm reminded of #1566 / #1832 / https://github.com/tidyverse/style/issues/160:

---

The issue in arises from the difference between AST-based parsers (`lintr`, `styler`, and `air`) and editor line heuristics (RStudio `Ctrl + I`) when handling unparenthesized binary operators of mixed precedence.

```r
is.null(a) ||
  length(a) == 1 &&
  is.na(a)
```

---

### The Concrete Discrepancy

When this code is formatted by the various tools:

* `lintr::indentation_linter()`: Expects 4 spaces on line 3 (`is.na(a)`).
* `styler::style_text()`: Formats line 3 with 4 spaces.
* `air format` (v0.11.0): Formats line 3 with 4 spaces.
* RStudio Auto-Indent (`Ctrl + I`): Inserts 2 spaces on line 3.

---

### Why `lintr`, `styler`, and `air` All Produce 4 Spaces

All three tools parse the code into an Abstract Syntax Tree (AST) following R's official grammar rules:

In base R grammar (`?Syntax`), `&&` has higher precedence than `||`. Therefore, `is.null(a) || length(a) == 1 && is.na(a)` parses as:

```r
is.null(a) || (length(a) == 1 && is.na(a))
```

---

### Why RStudio Auto-Indent (`Ctrl + I`) Produces 2 Spaces

RStudio's interactive editor indent engine does not build a full AST with operator precedence while editing. Instead, it uses a line-by-line lexical heuristic:
1. When evaluating line 3, it inspects line 2 (`length(a) == 1 &&`).
2. It detects a trailing continuation operator (`&&`).
3. It indents line 3 by a single indent level (+2 spaces) relative to the starting column of line 2, resulting in 2 spaces.

---

### Operator Order Dependency

When the operator order is reversed so that `&&` precedes `||`:

```r
length(a) == 1 &&
  is.na(a) ||
  is.null(a)
```

1. In R's grammar, this parses left-associatively as `(length(a) == 1 && is.na(a)) || is.null(a)`.
2. Line 1's `&&` only spans lines 1–2 (adding +2 to line 2).
3. Line 2's `||` only spans lines 2–3 (adding +2 to line 3).
4. **All tools (`lintr`, `styler`, `air`, and RStudio) agree on 2 spaces for both line 2 and line 3.**
